kyo-core/jvm/src/main/scala/kyo/Path.scala

```diff
@@ -339,22 +354,55 @@ object Path:
 
     type Part = String | Path
 
-    def apply(path: List[Part]): Path =
-        def loop(path: List[Part], acc: List[String]): List[String] =
-            path match
-                case _: Nil.type =>
-                    acc.reverse
-                case h :: t =>
-                    h match
-                        case h: String =>
-                            loop(t, h.split('/').filter(_.nonEmpty).toList.reverse ::: acc)
-                        case h: Path =>
-                            loop(h.path ::: t, acc)
-        new Path(loop(path, Nil))
+    def apply(parts: List[Part]): Path =
```

Comment:
I had issues with this method returning paths without the initial `/` and not handling `null` parts.